package.json

```diff
@@ -13,8 +13,7 @@
     "check": "node script/check-code-style.js",
     "prepublishOnly": "script/publish",
     "lint": "eslint . --ext .ts,.tsx,.js --fix",
-    "build": "npm install && node script/barrelize.js && tsc && npm run copy",
-    "doc": "npm run build && node node_modules/typedoc/bin/typedoc --options typedoc.js",
+    "build": "npm install && node script/barrelize.js && tsc && npm run copy && node node_modules/typedoc/bin/typedoc --options typedoc.js",
```

Comment:
Adding this to every build is going to slow things down during development. I would suggest creating `test:travis` which would do `test:retry` followed by `doc` and then use `test:travis` in [.travis.yml](https://github.com/aws/amazon-chime-sdk-js/blob/master/.travis.yml). If we are going to add it to the travis build, then we need to also confirm (via script) that regenerating docs produced no git diff. This will help catch cases where a PR changes documentation but does not include the generated documentation changes as well as provide extra assurance that checked-in documentation mirrors the inline code documentation.

`doc` also needs to make sure that any files that no longer exist after regenerating get removed. This could be done by wrapping the doc gen like this: `git rm -rf docs && ...typedoc... && git add docs`.

Reply:
Add predoc and postdoc to remove the previous docs folder and  update the doc link to point to master